nasl/nasl_crypto2.c

```diff
@@ -1547,7 +1547,8 @@ nasl_open_stream_cipher (lex_ctxt *lexic, int cipher, int mode)
 
   if (!key || keylen <= 0)
     {
-      nasl_perror (lexic, "Syntax: encrypt_data: Missing data or key argument");
+      nasl_perror (lexic, "Syntax: open_stream_cipher (called from "
+                          "open_rc4_cipher): Missing key argument");
```

Comment:
Despite this function currently is being used only with RC4, the message should not include the cipher name hardcoded, since the function can handle other stream cipher than RC4. If you need some more info, I suggest to include the cipher name in the message, which will give you a hint about the function making the call.
```suggestion
nasl_perror (lexic, "Syntax: open_stream_cipher (called with cipher: %d): "
                    "Missing key argument", gcry_cipher_algo_name (cipher));
```

Reply:
Is it possible to somehow pass the name of the calling function to this function here? IMHO the `called with cipher` is far from being optimal because we still don't know which function was exactly called here and can only guess it's name from the cipher then.

Reply:
Yes, it is possible. You can modify the function signature to receive a new parameter with the caller function name.